src/main/java/com/hms_networks/americas/sc/extensions/logging/Logger.java

```diff
@@ -80,6 +82,17 @@ public class Logger {
   /** Queue of unprinted logs. */
   private static LogQueue logQueue;
 
+  /**
+   * List for tracking whether a specific log has been outputted. This is used to prevent spamming
+   * the log with the same message.
+   *
+   * <p>If the log has already been outputted, its key will be in the list, otherwise it will not be
+   * present.
+   *
+   * @since 1.15.9
+   */
+  private static final List logOnceList = new ArrayList(); // List<String>
```

Comment:
Should this be implemented as a FIFO instead to protect against this list getting too large?

Reply:
I don't know what you mean by using a FIFO here. 

This was implemented with the reduction of memory in mind, thus the reason for storing *only* the key and not the body of the log message.

Reply:
If an application were to use this functionality in a way that is not in the spirt of the implementation it may generate up to an infinite amount of new keys over a a long enough runtime. 

Example: (this would be bound by the amount of values possible for whatever datatype tag is.
`Logger.LOG_WARN_ONCE(tag.valuestring(), tagValueOutOfBounds);`

if it was implemented as a fifo you could allow the logOnce list to grow to a set amount of entries, then trim the oldest so that even if the mechanism was abused you would not have memory issues.

Reply:
This could be implemented, but it was outside my original scope and makes the implementation less simple/straightforward.

This also checks for/guards against a case that I think is an application-level concern. We don't currently impose limits on other ArrayLists, including those used for retry messages, queued payloads, response error messages, etc., in our connectors, most of which are certain to be larger and more volatile.

Reply:
This is a public library and as such i think we should aim to provide the most stable library solution we can for users. I dont expect users of the library to dig into the source to attempt to find potential issues/bugs. 

I do not think it would be unreasonable to have a sane default, then allow for limit adjustment via a method
`Logger.setOnceMessageListSize(100)`
`Logger.setOnceMEssageListSize(-1) //no limit`

We probably should look into other uses of arraylist and see if we can improve the user experience in terms of noting in the javadocs implications of growing the list too big and/or setting limits which could be modifiable by the application if needed. 

I even think generating exceptions for out of size conditions would not be unreasonable.

Reply:
It's a good idea to cap the size to prevent resource issues, but we can add that later. It appears the aim of this PR is to address missing configuration options.

Reply:
I agree that this is a good idea, and is really applicable to many components in the library. I've created issue #86 to investigate and address this at a larger scale in the future.